src/util/child_common.c

```diff
@@ -40,8 +40,28 @@ struct sss_child_ctx {
     int child_status;
     sss_child_sigchld_callback_t cb;
     void *pvt;
+    struct sss_child_ctx **pvt_watch;
 };
 
+static void cancel_pvt_watch(struct sss_child_ctx *ctx)
+{
+    if (ctx->pvt_watch != NULL) {
+        talloc_set_destructor(ctx->pvt_watch, NULL);
+        talloc_free(ctx->pvt_watch);
+        ctx->pvt_watch = NULL;
+    }
+}
+
+static int pvt_watch_destructor(struct sss_child_ctx **watch)
+{
+    if ((watch != NULL) && (*watch != NULL)) {
+        (*watch)->cb = NULL;
```

Comment:
If `watch` is being freed, why do we want to set its fields to `NULL`. I could understand it if the memory pointed by the fields was freed, but that's not happening.

Reply:
I understood it. I hadn't noticed that pvt_watch points to the parent context.